File: lib/aria/gtm_strategy.mjs
// lib/aria/gtm_strategy.mjs
// Go-To-Market Strategy Engine for Arkmurus
//
// Generates market entry strategies based on relationship tier,
// product-market fit, competitive landscape, and partnership angles.
// Injected into ARIA context when market entry discussions arise.

import { searchKnowledge } from './knowledge.mjs';
import { getContactsByCountry } from './contacts.mjs';

// ── Regional GTM Playbooks ───────────────────────────────────────────────────
const GTM_PLAYBOOKS = {
  // TIER 1: INCUMBENT — Leverage existing relationships
  INCUMBENT: {
    timeToFirstDeal: '1-3 months',
    strategy: 'Direct engagement — leverage existing MoD relationships',
    steps: [
      'Identify specific requirement from latest intel sweep',
      'Match requirement to 2-3 qualified OEMs',
      'Contact MoD procurement directorate directly (Portuguese)',
      'Present capability brief with OEM technical data',
      'Arrange OEM demonstration visit',
      'Submit formal proposal with pricing + training + spares package',
      'Navigate end-user certificate and export licence',
      'Close — typical commission 8-12%',
    ],
    partnerNeeded: false,
    localAgentNeeded: false,
    keyRisk: 'Complacency — competitors are entering our markets. Stay proactive.',
  },
  // TIER 2: ESTABLISHED — Competitive positioning
  ESTABLISHED: {
    timeToFirstDeal: '3-6 months',
    strategy: 'Competitive bid — differentiate on broker value-add',
    steps: [
      'Monitor procurement portals and defence news for tender announcements',
      'Pre-qualify with MoD through existing contacts',
      'Identify which OEMs are already bidding (competitive intelligence)',
      'Position Arkmurus as multi-OEM assembler (package what no single OEM offers)',
      'Submit proposal with compliance + offset management as differentiator',
      'Offer training/sustainment package that OEM direct sales skip',
      'Follow up within 48 hours of every interaction',
    ],
    partnerNeeded: false,
    localAgentNeeded: 'Recommended for sustained presence',
    keyRisk: 'Established competitors have deeper relationships. Win on service quality.',
  },
  // TIER 3: DEVELOPING — Build relationships first
  DEVELOPING: {
    timeToFirstDeal: '6-12 months',
    strategy: 'Relationship-first — invest before expecting returns',
    steps: [
      'Attend regional defence exhibition (DSEI, ShieldAfrica, AAD)',
      'Identify and approach Defence Attaché at UK embassy/high commission',
      'Request introductory meeting with MoD procurement directorate',
      'Present Arkmurus credentials and OEM partnerships',
      'Offer free intelligence briefing on relevant market/product trends',
      'Identify specific near-term procurement opportunity',
      'Partner with an established OEM that needs local representation',
      'Build relationship over 2-3 meetings before commercial discussion',
      'First deal likely small (training, spares) — build trust before major equipment',
    ],
    partnerNeeded: 'OEM partnership essential for credibility',
    localAgentNeeded: 'Essential — cannot operate without local presence',
    keyRisk: 'Investment without guaranteed return. Pick battles carefully.',
  },
  // TIER 4: COLD ENTRY — Partnership is the only way in
  COLD_ENTRY: {
    timeToFirstDeal: '12-24 months',
    strategy: 'Partnership-led — find the right OEM or local firm first',
    steps: [
      'Research: which OEMs are already active in this market?',
      'Approach OEMs that need Africa/Lusophone expertise in exchange for market access',
      'Propose teaming agreement: Arkmurus brings compliance + Africa relationships, OEM brings product',
      'Identify local agent/representative (mandatory in most markets)',
      'Attend the major regional exhibition (IDEX for ME, DSA for SE Asia, EUROSATORY for Europe)',
      'Request meeting with local procurement authority via Defence Attaché',
      'Start with advisory/consultancy role — build credibility before brokering',
      'Target offset/local content opportunity as entry point (high broker value)',
      'First deal may take 18+ months — plan budget accordingly',
    ],
    partnerNeeded: 'Mandatory — without a partner, you cannot compete',
    localAgentNeeded: 'Mandatory',
    keyRisk: 'High investment, uncertain return. Only enter if market opportunity score > 60.',
  },
};

// ── Market-specific entry intelligence ───────────────────────────────────────
const MARKET_ENTRY_INTEL = {
  'Angola': { tier: 'INCUMBENT', exhibition: 'FILDA (Luanda, Jul)', language: 'Portuguese', keyRelationship: 'Defence Attaché + MoD direct', bestOEM: 'Embraer (Lusophone), Paramount (vehicles), Baykar (UAVs)', offset: 'Informal — local content appreciated but not mandated' },
  'Mozambique': { tier: 'INCUMBENT', exhibition: 'FACIM (Maputo, Aug)', language: 'Portuguese', keyRelationship: 'SADC framework + EU SAMIM mission', bestOEM: 'Paramount (vehicles), Damen (naval), Embraer (air)', offset: 'None formal' },
  'Nigeria': { tier: 'ESTABLISHED', exhibition: 'None major — DSEI/ShieldAfrica', language: 'English', keyRelationship: 'MoD + individual service chiefs', bestOEM: 'Leonardo (trainers), Damen (naval), Turkish OEMs (competitive pricing)', offset: 'Local content via DICON partnership' },
  'Kenya': { tier: 'ESTABLISHED', exhibition: 'None major — DSEI', language: 'English', keyRelationship: 'MoD + AMISOM/ATMIS framework', bestOEM: 'Elbit (ISR), Paramount (vehicles), Airbus (helicopters)', offset: 'None formal' },
  'Indonesia': { tier: 'COLD_ENTRY', exhibition: 'Indo Defence (Jakarta, Nov, biennial)', language: 'English/Indonesian', keyRelationship: 'Need Korean OEM partner (Hanwha/KAI already established)', bestOEM: 'Hanwha (vehicles/artillery), KAI (aircraft) — partner, not compete', offset: '35-85% mandatory — broker navigates' },
  'Philippines': { tier: 'COLD_ENTRY', exhibition: 'ADAS (Manila, Sep)', language: 'English', keyRelationship: 'US FMS dominant — position as alternative channel', bestOEM: 'KAI (FA-50 already sold), Elbit (ISR), Damen (naval)', offset: 'None formal' },
  'Saudi Arabia': { tier: 'COLD_ENTRY', exhibition: 'IDEX (Abu Dhabi, Feb) + WDS (Riyadh, Mar)', language: 'Arabic/English', keyRelationship: 'GAMI is gatekeeper — agent mandatory', bestOEM: 'European primes (Leonardo, Airbus) — need premium product for premium market', offset: '50-60% via GAMI — Arkmurus manages offset compliance' },
  'UAE': { tier: 'COLD_ENTRY', exhibition: 'IDEX (Abu Dhabi, Feb)', language: 'Arabic/English', keyRelationship: 'EDGE Group partnership essential', bestOEM: 'Partner with EDGE subsidiary + European OEM', offset: '60% Tawazun — broker value highest here' },
  'Senegal': { tier: 'DEVELOPING', exhibition: 'ShieldAfrica (Abidjan, Jan, biennial)', language: 'French', keyRelationship: 'French Defence Attaché + Ministère des Forces Armées', bestOEM: 'Nexter (French heritage), Turkish OEMs (price competitive)', offset: 'None formal' },
  'Ghana': { tier: 'DEVELOPING', exhibition: 'None major — DSEI', language: 'English', keyRelationship: 'MoD Burma Camp + UK defence relationship', bestOEM: 'Paramount (vehicles), Damen (naval)', offset: 'None formal' },
  'Ethiopia': { tier: 'DEVELOPING', exhibition: 'None — bilateral engagement', language: 'English/Amharic', keyRelationship: 'Post-Tigray reconstruction window — MoD receptive to new suppliers', bestOEM: 'Baykar (already sold TB2), Turkish/Chinese alternatives to Russian legacy fleet', offset: 'None formal' },
  'Poland': { tier: 'COLD_ENTRY', exhibition: 'MSPO (Kielce, Sep)', language: 'Polish/English', keyRelationship: 'Need Korean partner (K2/K9 mega-deals established relationship)', bestOEM: 'Hanwha, Hyundai Rotem — already in market, partner not compete', offset: 'EU/NATO procurement rules' },
  'Brazil': { tier: 'INCUMBENT', exhibition: 'LAAD (Rio, Apr, biennial)', language: 'Portuguese', keyRelationship: 'Deep Lusophone ties + Embraer relationship', bestOEM: 'Embraer (domestic champion), Iveco (Brazilian factory)', offset: '100% for deals >R$50M — broker essential for offset navigation' },
};

/**
 * Generate full GTM strategy for a market.
 */
export function generateGTMStrategy(market) {
  const intel = MARKET_ENTRY_INTEL[market];
  if (!intel) return null;

  const playbook = GTM_PLAYBOOKS[intel.tier];
  const contacts = getContactsByCountry(market);

  return {
    market,
    tier: intel.tier,
    playbook,
    exhibition: intel.exhibition,
    language: intel.language,
    keyRelationship: intel.keyRelationship,
    bestOEM: intel.bestOEM,
    offset: intel.offset,
    contacts: contacts.slice(0, 5),
    timeToFirstDeal: playbook.timeToFirstDeal,
  };
}

/**
 * Context injection for ARIA when GTM discussions arise.
 */
export function getGTMContext(query) {
  const q = (query || '').toLowerCase();
  const triggers = ['go to market', 'market entry', 'enter market', 'how do we get into',
    'strategy for', 'break into', 'expand to', 'start in', 'enter the', 'grow into'];
  if (!triggers.some(t => q.includes(t))) return '';

  const markets = Object.keys(MARKET_ENTRY_INTEL);
  const market = markets.find(m => q.includes(m.toLowerCase()));
  if (!market) return '';

  const strategy = generateGTMStrategy(market);
  if (!strategy) return '';

  return '\n\n[GO-TO-MARKET STRATEGY — ' + market + ' (' + strategy.tier + ')]\n' +
    'Time to first deal: ' + strategy.timeToFirstDeal + '\n' +
    'Strategy: ' + strategy.playbook.strategy + '\n' +
    'Exhibition: ' + strategy.exhibition + '\n' +
    'Language: ' + strategy.language + '\n' +
    'Key relationship: ' + strategy.keyRelationship + '\n' +
    'Best OEMs: ' + strategy.bestOEM + '\n' +
    'Offset: ' + strategy.offset + '\n' +
    'Partner needed: ' + strategy.playbook.partnerNeeded + '\n' +
    'Local agent: ' + strategy.playbook.localAgentNeeded + '\n' +
    'Key risk: ' + strategy.playbook.keyRisk + '\n' +
    'Steps:\n' + strategy.playbook.steps.map((s, i) => (i + 1) + '. ' + s).join('\n');
}

File: aria_service/intel/gtm_strategy.py
"""
Go-To-Market Strategy Engine.
Ported from lib/aria/gtm_strategy.mjs.
"""
from __future__ import annotations
from .engine_wiring import wire_failure

from .contacts import search_contacts

# ── GTM Playbooks by Tier ────────────────────────────────────────────────────

PLAYBOOKS = {
    "INCUMBENT": {
        "description": "Direct engagement via existing MoD relationships",
        "local_agent": "NO",
        "partner_needed": "NO",
        "typical_cycle": "1-3 months",
        "key_risk": "Complacency — competitors entering your markets",
        "entry_points": ["Direct MoD contact", "Defence attaché channel", "CPLP framework"],
    },
    "ESTABLISHED": {
        "description": "Competitive bid, differentiate on broker value-add",
        "local_agent": "Recommended",
        "partner_needed": "NO",
        "typical_cycle": "3-6 months",
        "key_risk": "Established competitors have deeper relationships",
        "entry_points": ["Monitor tender portals", "Pre-qualify with contacts", "Exhibition meetings"],
    },
    "DEVELOPING": {
        "description": "Relationship-first before commercial pitch",
        "local_agent": "Essential",
        "partner_needed": "OEM partnership essential",
        "typical_cycle": "6-12 months",
        "key_risk": "Investment without guaranteed return",
        "entry_points": ["Regional expos (DSEI, ShieldAfrica, AAD)", "Defence Attaché introductions"],
    },
    "COLD_ENTRY": {
        "description": "Partnership-led entry (find the right OEM or local firm first)",
        "local_agent": "Mandatory",
        "partner_needed": "Mandatory",
        "typical_cycle": "12-24 months",
        "key_risk": "High investment, uncertain return — only enter if opportunity score > 60",
        "entry_points": ["OEM teaming agreement", "Local industry JV", "Government-sponsored expo"],
    },
}

# ── Market-to-Tier mapping ───────────────────────────────────────────────────

MARKETS = {
    "Angola": {"tier": "INCUMBENT", "exhibition": "FILDA (Luanda, Jul)", "language": "Portuguese", "best_oem": "Embraer, Paramount", "offset": "Informal", "key_relationship": "SIMPORTEX Director"},
    "Mozambique": {"tier": "INCUMBENT", "exhibition": "Maputo Defence Forum", "language": "Portuguese", "best_oem": "Paramount, Damen", "offset": "Informal", "key_relationship": "FADM Procurement Director"},
    "Brazil": {"tier": "INCUMBENT", "exhibition": "LAAD (Rio, Apr)", "language": "Portuguese", "best_oem": "Embraer", "offset": "100% >R$50M", "key_relationship": "Defence Cooperation Coordinator"},
    "Nigeria": {"tier": "ESTABLISHED", "exhibition": "DSEI/ShieldAfrica", "language": "English", "best_oem": "Leonardo, Damen, Turkish", "offset": "DICON partnership", "key_relationship": "DICON Director"},
    "Kenya": {"tier": "ESTABLISHED", "exhibition": "DSEI/AAD", "language": "English", "best_oem": "Various", "offset": "Informal", "key_relationship": "KDF Procurement Director"},
    "Guinea-Bissau": {"tier": "INCUMBENT", "exhibition": "None regular", "language": "Portuguese", "best_oem": "Light equipment", "offset": "None", "key_relationship": "Family connections + CPLP"},
    "Senegal": {"tier": "DEVELOPING", "exhibition": "ShieldAfrica (Abidjan)", "language": "French", "best_oem": "Nexter, Airbus", "offset": "Informal", "key_relationship": "Defence Attaché"},
    "Ghana": {"tier": "DEVELOPING", "exhibition": "ShieldAfrica", "language": "English", "best_oem": "Paramount, Otokar", "offset": "Informal", "key_relationship": "MoD Procurement"},
    "Ethiopia": {"tier": "DEVELOPING", "exhibition": "None regular", "language": "English", "best_oem": "Various", "offset": "Informal", "key_relationship": "MoD Directorate"},
    "Indonesia": {"tier": "COLD_ENTRY", "exhibition": "Indo Defence (Nov)", "language": "English/Indonesian", "best_oem": "Hanwha, KAI (partner)", "offset": "35-85% mandatory", "key_relationship": "OEM local partner"},
    "Philippines": {"tier": "COLD_ENTRY", "exhibition": "ADAS (Manila)", "language": "English", "best_oem": "Various (diversifying from US FMS)", "offset": "Informal", "key_relationship": "DND Procurement"},
    "Saudi Arabia": {"tier": "COLD_ENTRY", "exhibition": "IDEX/WDS", "language": "Arabic/English", "best_oem": "European primes", "offset": "50-60% via GAMI", "key_relationship": "GAMI gatekeeper"},
    "UAE": {"tier": "COLD_ENTRY", "exhibition": "IDEX", "language": "Arabic/English", "best_oem": "EDGE subsidiary + OEM", "offset": "60% Tawazun", "key_relationship": "EDGE Group"},
    "India": {"tier": "COLD_ENTRY", "exhibition": "DefExpo (Feb)", "language": "English", "best_oem": "European non-ITAR", "offset": "30%+ Make in India", "key_relationship": "MoD Acquisition Wing"},
    "Poland": {"tier": "COLD_ENTRY", "exhibition": "MSPO Kielce (Sep)", "language": "English/Polish", "best_oem": "Hanwha, Lockheed, BAE", "offset": "Industrial cooperation", "key_relationship": "Armament Agency"},
    "Egypt": {"tier": "COLD_ENTRY", "exhibition": "EDEX (Dec)", "language": "Arabic/English", "best_oem": "European + Korean", "offset": "Arab Industrial Organisation", "key_relationship": "MoD Procurement"},
    "Morocco": {"tier": "COLD_ENTRY", "exhibition": "None regular", "language": "French/Arabic", "best_oem": "French/US", "offset": "Informal", "key_relationship": "Royal Armed Forces HQ"},
    "Colombia": {"tier": "COLD_ENTRY", "exhibition": "ExpoDefensa (Dec)", "language": "Spanish", "best_oem": "US/Israeli/European", "offset": "Informal", "key_relationship": "MoD Acquisition"},
    "Australia": {"tier": "COLD_ENTRY", "exhibition": "Land Forces (Oct)", "language": "English", "best_oem": "AUKUS partners", "offset": "AIC programme", "key_relationship": "CASG"},
    "Japan": {"tier": "COLD_ENTRY", "exhibition": "DSEI Japan (Mar)", "language": "English/Japanese", "best_oem": "US alliance only", "offset": "Licensed production", "key_relationship": "ATLA"},
    "Turkey": {"tier": "COLD_ENTRY", "exhibition": "IDEF Istanbul (May)", "language": "English/Turkish", "best_oem": "Domestic SSB", "offset": "Co-production", "key_relationship": "SSB (Defence Industry Agency)"},
    "Vietnam": {"tier": "COLD_ENTRY", "exhibition": "None regular", "language": "English/Vietnamese", "best_oem": "Israeli/Indian/Korean", "offset": "Informal", "key_relationship": "General Dept of Defence Industry"},
    "Romania": {"tier": "COLD_ENTRY", "exhibition": "BSDA Bucharest (Jun)", "language": "English/Romanian", "best_oem": "NATO-standard", "offset": "Industrial cooperation", "key_relationship": "MoD DPA"},
    "South Korea": {"tier": "COLD_ENTRY", "exhibition": "ADEX Seoul (Oct)", "language": "English/Korean", "best_oem": "Domestic DAPA", "offset": "Technology transfer", "key_relationship": "DAPA"},
}


# ── Public API ───────────────────────────────────────────────────────────────

def generate_gtm_strategy(market: str) -> dict | None:
    """Generate GTM strategy for a market."""
    info = MARKETS.get(market)
    if not info:
        # Fuzzy
        ml = market.lower()
        for k, v in MARKETS.items():
            if ml in k.lower():
                info = v
                market = k
                break
    if not info:
        return None

    playbook = PLAYBOOKS.get(info["tier"], PLAYBOOKS["DEVELOPING"])
    contacts = search_contacts(market)[:3]

    return {
        "market": market,
        "tier": info["tier"],
        "playbook": playbook,
        "exhibition": info["exhibition"],
        "language": info["language"],
        "keyRelationship": info["key_relationship"],
        "bestOEM": info["best_oem"],
        "offset": info["offset"],
        "contacts": contacts,
        "timeToFirstDeal": playbook["typical_cycle"],
    }


def get_gtm_context(query: str) -> str:
    """Synchronous prompt injection — triggers on GTM-related queries."""
    ql = query.lower()
    gtm_kws = ["go to market", "market entry", "break into", "enter market", "gtm", "how to sell"]
    if not any(k in ql for k in gtm_kws):
        return ""
    for market in MARKETS:
        if market.lower() in ql:
            result = generate_gtm_strategy(market)
            if not result:
                return ""
            pb = result["playbook"]
            lines = [
                f"\n[GO-TO-MARKET STRATEGY — {market} (Tier: {result['tier']})]",
                f"Strategy: {pb['description']}",
                f"Local agent: {pb['local_agent']} | Partner: {pb['partner_needed']}",
                f"Cycle: {pb['typical_cycle']} | Exhibition: {result['exhibition']}",
                f"Key risk: {pb['key_risk']}",
                f"Best OEM: {result['bestOEM']} | Key relationship: {result['keyRelationship']}",
            ]
            return "\n".join(lines)
    # R-F996 — wire to brain
    from .engine_wiring import wire_success, wire_failure
    wire_success(
        module="gtm_strategy",
        summary="Get Gtm Context",
        source_id="gtm_strategy:R-F996",
    )

    return ""

# R-F2119 §21a — wire failure handler for gtm_strategy
try:
    wire_failure(module="gtm_strategy", detail="module shutdown",
                gap_type="engine_failure", source="gtm_strategy:shutdown")
except Exception:
    pass
